src/Dotnet.Script.Core/ScriptPublisher.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Dotnet.Script.DependencyModel.Environment;
+using Dotnet.Script.DependencyModel.Logging;
+using Dotnet.Script.DependencyModel.Process;
+using Dotnet.Script.DependencyModel.ProjectSystem;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Scripting;
+using Microsoft.CodeAnalysis.Scripting.Hosting;
+using System;
+using System.IO;
+using System.Reflection;
+
+namespace Dotnet.Script.Core
+{
+    public class ScriptPublisher
+    {
+        const string AssemblyName = "scriptAssembly";
+
+        private readonly ScriptProjectProvider _scriptProjectProvider;
+        private readonly ScriptCompiler _scriptCompiler;
+        private readonly ScriptConsole _scriptConsole;
+        private readonly ScriptEnvironment _scriptEnvironment;
+
+        public ScriptPublisher(ScriptProjectProvider scriptProjectProvider, ScriptCompiler scriptCompiler,
+            ScriptConsole scriptConsole)
+        {
+            _scriptProjectProvider = scriptProjectProvider ?? throw new ArgumentNullException(nameof(scriptProjectProvider));
+            _scriptCompiler = scriptCompiler ?? throw new ArgumentNullException(nameof(scriptCompiler));
+            _scriptConsole = scriptConsole ?? throw new ArgumentNullException(nameof(scriptConsole));
+            _scriptEnvironment = ScriptEnvironment.Default;
+        }
+
+        public ScriptPublisher(LogFactory logFactory, ScriptCompiler scriptCompiler)
+            : this
+            (
+                new ScriptProjectProvider(logFactory),
+                scriptCompiler,
+                ScriptConsole.Default
+            )
+        {
+        }
+
+        public void CreateExecutable(ScriptContext context, LogFactory logFactory)
+        {
+            var tempProjectPath = _scriptProjectProvider.CreateProjectForScriptFile(context.FilePath);
+            var tempProjectDirecory = Path.GetDirectoryName(tempProjectPath);
+
+            var scriptAssemblyPath = CreateScriptAssembly(context, tempProjectDirecory);
+
+            var projectFile = new ProjectFile(File.ReadAllText(tempProjectPath));
+            // todo: grab version in a better way?
+            projectFile.AddPackageReference(new PackageReference("Microsoft.CodeAnalysis.Scripting", "2.8.2", PackageOrigin.ReferenceDirective));
```

Comment:
I think it's ok to hardcode a version. You could technically omit it, with .net core SDK 2.1.300 it would just pull the latest but it might conflict with something. Let's make a global constant for the version